src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -703,12 +700,6 @@ DeferErrorIfUnsupportedSubqueryPushdown(Query *originalQuery,
 		{
 			return error;
 		}
-
-		error = DeferErrorIfUnsupportedFilters(subquery);
```

Comment:
This check also applies for non-set operations as well. So, we should add something like the following:
```SQL

SELECT *
FROM
  (SELECT *
   FROM
     (SELECT user_id,
             sum(value_2) AS counter
      FROM events_table
      WHERE user_id = 16
      GROUP BY user_id) AS foo,
     (SELECT user_id,
             sum(value_2) AS counter
      FROM events_table
      WHERE user_id = 15
      GROUP BY user_id) AS bar
   WHERE foo.user_id = bar.user_id ) AS baz;
```

(Well actually the above query is pretty useless, but, the following which is slightly more useful does seem to work on the master branch, which is probably a bug)
```SQL

SELECT *
FROM
  (SELECT *
   FROM
     (SELECT user_id,
             sum(value_2) AS counter
      FROM events_table
      WHERE user_id = 16
        OR user_id = 17
      GROUP BY user_id) AS foo,

     (SELECT user_id,
             sum(value_2) AS counter
      FROM events_table
      WHERE user_id = 15
        OR user_id = 16
      GROUP BY user_id) AS bar
   WHERE foo.user_id = bar.user_id ) AS baz;


```